src/main/java/org/commcare/formplayer/session/FormSession.java

```diff
@@ -108,7 +111,9 @@ public FormSession(SerializableFormSession session,
 
         this.sandbox = restoreFactory.getSandbox();
         this.windowWidth = windowWidth;
-
+        if (commCareSession != null && commCareSession.getCurrentEntry() != null) {
+            this.formTitleLocaleKey = commCareSession.getCurrentEntry().getText().getArgument();
```

Comment:
think we need to use`getText().evaluate()` which already seems to have [locale support](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/commcare/suite/model/Text.java#L213), so you might be able to use it's value directly without doing any further locale processing.

Reply:
HI @shubham1g5 I explored this approach before and revisited after seeing your comment. The issue is that it seems that `getText().evaluate()` [fetches the values using the default locale](https://github.com/dimagi/commcare-core/blob/25a931722f4441ae01c7d4531ba19a99ce9be96f/src/main/java/org/commcare/suite/model/Text.java#L221C24-L221C36) and not the current language

Reply:
I think that takes into account whatever locale is set to the Localiser, saying as we use ` Localization.get()` at other several places in the app to get the translation.

Reply:
Yes, but when we change language in the middle of a form session, it doesn't change the global locale, just the [form-level locale](https://github.com/dimagi/formplayer/blob/bb7cdbb2db7418834938e94222167be2aff98561/src/main/java/org/commcare/formplayer/session/FormSession.java#L636). So calling `getText().evaluate() `continues to return the value in the default language, right?